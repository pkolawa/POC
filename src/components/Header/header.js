import React from "react";
import { withRouter } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import HomeIcon from "@material-ui/icons/Home";

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  HomeButton: {
    marginRight: theme.spacing(2),
  },
}));

function Header(props) {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="fixed">
        <Toolbar>
          <IconButton
            edge="start"
            className={classes.HomeButton}
            color="inherit"
            aria-label="Home"
            onClick={() => (props.history.push('/'))}
          >
            <HomeIcon />
          </IconButton>
          <Typography variant="h6" color="inherit">
            HTTP APP Console
          </Typography>
        </Toolbar>
      </AppBar>
    </div>
  );
}

export {Header}

export default withRouter(Header);