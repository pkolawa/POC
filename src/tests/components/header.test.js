import React from "react";
import { shallow } from "enzyme";
import { Header } from "../../components/Header/header";

test("Should Render Header", () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});
