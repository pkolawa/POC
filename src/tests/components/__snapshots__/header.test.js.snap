// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render Header 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(AppBar))
    position="fixed"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="Home"
        className="makeStyles-HomeButton-2"
        color="inherit"
        edge="start"
        onClick={[Function]}
      >
        <HomeIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        color="inherit"
        variant="h6"
      >
        HTTP APP Console
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
